src/yaml.rs

```diff
@@ -0,0 +1,65 @@
+//! Wrapper around serde_yaml which adheres to the YAML specification
```

Comment:
I think strictly speaking the --- are only mandatory if the file starts with directives, to delimit these from the actual content of the file.
So arguably serde_yaml adheres to the spec, even though I have to say that I do not agree with the upstream change.
Besides that, it might be worthwhile just adding one sentence about what exactly the difference is between this and serde_yaml to make it easier to understand.

Reply:
In YAML 1.0 and 1.1, a YAML stream can start with an "implicit document" (without dashes) or an "exlicit document" (with dashes), see https://yaml.org/spec/1.0/index.html#id2561559 and https://yaml.org/spec/1.1/current.html#id898785.

It is the same in YAML 1.2 but "implicit documents" are called "bare documents", see https://yaml.org/spec/1.2.2/#913-bare-documents.

So serde_yaml is compliant with the YAML specification. I will change my pull request.

Reply:
Is it worth opening an issue against yamllint for this?

Reply:
There is already a closed issue with this topic: adrienverge/yamllint#302